lib/ecto/adapters/postgres/connection.ex

```diff
@@ -1844,6 +1845,15 @@ if Code.ensure_loaded?(Postgrex) do
       [type, generated_expr(generated)]
     end
 
+    defp extract_opts(opts) do
+      with {:ok, from} <- Keyword.fetch(opts, :from),
+           {_type, from_opts} <- from do
+        from_opts
+      else
+        _ -> opts
+      end
+    end
```

Comment:
```suggestion
    defp extract_from_opts({_type, opts), do: opts
    defp extract_from_opts(_), do: []
```
I think we might need to do something like this. Otherwise if someone supplies something like this it will think they are equal:

`modify(:field, :string, size: 100, from: :string)`